seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java

```diff
@@ -317,12 +334,15 @@ public List<String> listTables(String databaseName)
 
     @Override
     public boolean tableExists(TablePath tablePath) throws CatalogException {
-        try {
-            return databaseExists(tablePath.getDatabaseName())
-                    && listTables(tablePath.getDatabaseName()).contains(getTableName(tablePath));
-        } catch (DatabaseNotExistException e) {
+        String databaseName = tablePath.getDatabaseName();
+        if (!databaseExists(databaseName)) {
             return false;
         }
+        if (EXCLUDED_SCHEMAS.contains(tablePath.getSchemaName())) {
+            return false;
+        }
+        return queryExists(
+                this.getUrlFromDatabaseName(databaseName), getTableWithConditionSql(tablePath));
```

Comment:
```suggestion
        String databaseName = tablePath.getDatabaseName();
        if (EXCLUDED_SCHEMAS.contains(tablePath.getSchemaName())) {
            return false;
        }
        return queryExists(
                this.getUrlFromDatabaseName(databaseName), getTableWithConditionSql(tablePath));
```
how about remove check database exist? It won't affect the result.